### PR TITLE
Migrate to typescript & prettier

### PR DESCRIPTION
**Pull request by crookedneighbor**
Initial PR

### CODE REVIEW

File: src/class-list.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+function _classesOf(element): string[] {
+  return element.className.trim().split(/\s+/);
+}
+
+export function add(element, ...args): void {
+  const toAdd = Array.from(args);
+  const className = _classesOf(element)
```
Comment (gesa) [inline src/class-list.ts]: this variable is redundant, i'd rather see this written as `element.className` =

File: src/class-list.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    .filter(function (classname) {
+      return toAdd.indexOf(classname) === -1;
+    })
```
Comment (gesa) [inline src/class-list.ts]: this should be an arrow function—beyond the `this` distinction, functions that only return are like the ideal use case. also array methods.

```suggestion
    .filter((classname) => toAdd.indexOf(classname) === -1)
```

**Review by gesa — CHANGES_REQUESTED** [review]
ts compiles to es5 anyway, so may as well update the code just generally ¯\_(ツ)_/¯ 

p.s. we gotta do something about these gitignore files. they're out of control in a mostly irrelevant way.

File: src/class-list.ts (hunk 1 above)
Comment (crookedneighbor) [reply src/class-list.ts]: :+1:

File: src/__tests__/class-list.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
+import { add, remove, toggle } from "../class-list";
+
+describe("classList", () => {
+  let element;
```
Comment (gesa) [inline src/__tests__/class-list.ts]: this one change and the whole thing is strict compliant

```suggestion
  let element: HTMLDivElement;
```

**Review by gesa — APPROVED** [review]

**Pull request merged**